extensions/wikia/UserLogin/js/FacebookLogin.js

```diff
@@ -287,6 +287,30 @@
 					}
 				}
 			});
+
+			this.initSignupFormValidation();
+		},
+
+		initSignupFormValidation: function () {
+			var validator,
+				wikiaForm = this.signupForm.wikiaForm,
+				inputs = wikiaForm.inputs,
+				inputsToValidate = ['username', 'password'];
+
+			if (inputs.email) {
+				inputsToValidate.push('email');
+			}
+
+			validator = new UserSignupAjaxValidation({
+				wikiaForm: wikiaForm,
+				inputsToValidate: inputsToValidate,
+				submitButton: inputs.submit
+			});
+
+			// Add validation on blur event for all inputs to validate
+			inputsToValidate.forEach(function (inputName) {
+				inputs[inputName].on('blur', validator.validateMappedInput.bind(validator));
```

Comment:
I guess that 'inputs' is an array of jQuery objects, so isn't there a chance to bind this validation to blur event only once? I mean, not like in this loop, but once using prepared, filtered jQuery collection.
It could be much more effective from performance point of view.